### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by autodidaddict**
This is going to grow over time. It's not done yet, so please just review it based on whether it can be committed as a foundation for a larger sample that will definitely have many changes.

### CODE REVIEW

**Comment by autodidaddict** [comment]
As an FYI, some of this stuff is used as a source for tutorials and guides in the documentation.

File: ecommerce/payments-interface/Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+all: deps codegen build
+
+deps:
+
+codegen:
+	wapc generate codegen.yaml
+
+build:
+	cargo build
```
Comment (brooksmtownsend) [inline ecommerce/payments-interface/Makefile:11]: This can be removed

File: ecommerce/payments-interface/payments.widl [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+type AuthorizePaymentResponse {
+    success: bool
+    auth_code: string?
+    fail_reason: string?
+}
+
+type CompletePaymentRequest {
+    auth_code: string
+    request: AuthorizePaymentRequest
+}
```
Comment (brooksmtownsend) [inline ecommerce/payments-interface/payments.widl:25]: These snake_case variables should be camelCase right? The latest update to `wapc` cli corrects the parameter naming issue

**Review by brooksmtownsend — CHANGES_REQUESTED** [review]
I think this is a great foundation for a payments example. `ecommerce` is a good use case for wasmCloud so it will be a good example to have in here.

**Review by brooksmtownsend — APPROVED** [review]

**Pull request merged**